src/components/AddPhoto.jsx

```diff
@@ -14,12 +15,32 @@ function AddPhoto({ refetchPhotos }) {
 
   return (
     <div className="input-container">
-      <TextField id="outlined-basic" label="Image URL" variant="outlined" value={url} onChange={(e) => {
+      <TextField
+        id="outlined-basic"
+        label="Image URL"
+        variant="outlined"
+        value={url}
+        onChange={(e) => {
           setUrl(e.target.value);
-        }}/> 
-        <br/> <br/>
-        <GitHub/> <br/><br/>
-      <Button variant="contained" color="success" onClick={handleSubmit}>Send</Button> <br />
+        }}
+      />
+      <br /> <br />
+      <GitHub /> <br />
+      <br />
+      <Button
+        variant="contained"
+        color="success"
+        onClick={() => {
+          if (new URL(url)) {
+            handleSubmit();
+          } else {
+            throw new Error("Invalid URL");
```

Comment:
URL throws an error when the url is invalid, you should handle that behavior properly with try/catch, create a new function for that